docs/examples/perceptualdecisionmaking.ipynb

```diff
@@ -0,0 +1,434 @@
+    "rewards = {\n",
+    "    \"abort\": -0.1,\n",
+    "    \"correct\": +1.0,\n",
+    "    \"fail\": 0.0\n",
+    "}\n",
+    "timing = {\n",
+    "    \"fixation\": TruncExp(600, 400, 700),\n",
+    "    \"stimulus\": 2000,\n",
+    "    \"delay\": 0,\n",
+    "    \"decision\": 100,\n",
+    "}\n",
+    "sigma = 1.0 # Standard deviation of the Gaussian noise in the ring representation\n",
+    "\n",
+    "kwargs = {\n",
+    "    \"dt\": dt,\n",
+    "    \"dim_ring\": dim_ring,\n",
+    "    \"rewards\": rewards,\n",
+    "    \"timing\": timing,\n",
+    "    \"sigma\": sigma,\n",
+    "    \"abort\": abort,\n",
+    "}\n",
+    "block_dur = (20, 100) # Extremes of the block duration in milliseconds\n",
+    "probs = [[0.2, 0.8], [0.8, 0.2]] # Probabilities of choosing left or right in the two blocks\n",
+    "\n",
+    "# Create and wrap the environment\n",
+    "task = \"PerceptualDecisionMaking-v0\"\n",
+    "env = gym.make(task, **kwargs)\n",
+    "env = ReactionTime(env, end_on_stimulus=True)\n",
+    "env = PassReward(env)\n",
+    "env = PassAction(env)\n",
+    "env = SideBias(env, probs=probs, block_dur=block_dur)\n",
+    "\n",
+    "# Print environment specifications\n",
+    "print(\"Trial timing (in milliseconds):\")\n",
+    "print(env.timing)\n",
+    "\n",
+    "print(\"\\nObservation space structure:\")\n",
+    "print(env.observation_space)\n",
+    "\n",
+    "print(\"\\nAction space structure:\")\n",
+    "print(env.action_space)\n",
+    "print(\"Action mapping:\")\n",
+    "print(env.action_space.name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.1.3 Random Agent Behavior\n",
+    "\n",
+    "Let's now plot the behavior of a random agent on the task. The agent will randomly choose between the two options (left/right), and we will visualize its behavior over 5 trials. We will also plot the reward received by the agent at each time step, as well as the performance on each trial. Note that performance is only defined at the end of a trial: it is 1 if the agent made the correct choice, and 0 otherwise.\n",
+    "\n",
+    "To keep track of the agent's behavior, we will use the `Monitor` wrapper, which monitors training by:\n",
+    "\n",
+    "- Tracking and saving behavioral data (rewards, actions, observations) every `sv_per` steps.\n",
+    "- Generating visualization figures during training if` sv_fig=True`.\n",
+    "- Providing progress information if `verbose=True`.\n",
+    "\n",
+    "Here, we’ll use the wrapper solely to compute the agent’s performance, but later it will help us assess learning and save intermediate results.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Visualize example trials\n",
+    "fig = ngym.utils.plot_env(\n",
```

Comment:
slightly outside of the scope of this PR: it would be nice to have a light/dashed horizontal line in the "Rewards" graph at 0, to make it visually clearer that it goes negative on "abort".

Reply:
I don't have strong opinions about this, feel free to open an issue in case :)